packages/gatsby-theme-carbon/src/components/ResourceCard/ResourceCard.js

```diff
@@ -1,112 +1,184 @@
+  });
+
+  // These are all the icons available to the resource card.
+  const icons = (
+    <>
+      {actionIcon === 'launch' && !disabled ? (
+        <Launch20 aria-label="Open resource" />
+      ) : null}
+      {actionIcon === 'arrowRight' && !disabled ? (
+        <ArrowRight20 aria-label="Open resource" />
+      ) : null}
+      {actionIcon === 'download' && !disabled ? (
+        <Download20 aria-label="Download" />
+      ) : null}
+      {actionIcon === 'email' && !disabled ? (
+        <Email20 aria-label="Email" />
+      ) : null}
+      {actionIcon === 'calendar' && !disabled ? (
+        <Calendar20 aria-label="Calendar" />
+      ) : null}
+      {actionIcon === 'disabled' || disabled === true ? (
+        <Error20 aria-label="disabled" />
+      ) : null}
+    </>
+  );
+
+  // This determines the size of the title prop. productive-heading-03 for default and body-long-02 for Mini.
+  const titleSize = (
+    <>
+      {type === 'default'
+        ? title && (
+            <h4 className={`${prefix}--resource-card__title`}>{title}</h4>
+          )
+        : null}
+      {type === 'mini'
+        ? title && (
+            <h4 className={`${prefix}--resource-card__mini-title`}>{title}</h4>
+          )
+        : null}
+    </>
+  );
+
+  // This determines how and where the icons and image (children) are placed on the page.
+  const iconsAndImages = (
+    <>
+      {type === 'default'
+        ? icons &&
+          children && (
+            <>
+              <div className={`${prefix}--resource-card__icon--img`}>
+                {children}
+              </div>
+              <div className={`${prefix}--resource-card__icon--action`}>
+                {icons}
+              </div>
+            </>
+          )
+        : null}
+      {type === 'mini' ? (
+        children === undefined ? (
+          <div className={`${prefix}--resource-card__mini-icon--action`}>
+            {icons}
           </div>
-        </div>
-      </div>
+        ) : (
+          <div className={`${prefix}--resource-card__mini-icon--img`}>
+            {children}
+          </div>
+        )
+      ) : null}
+    </>
+  );
+
+  // This holds the subtitle, title, icons and images.
+  const cardContent = (
+    <>
+      {subTitle && (
+        <h5 className={`${prefix}--resource-card__subtitle`}>{subTitle}</h5>
+      )}
+      {titleSize}
+      {iconsAndImages}
+    </>
+  );
+
+  // This determines if the link that is supplied is staying internal to the site or if it's leaving the site.
+  let cardContainer;
+  if (disabled === true || href === undefined) {
+    cardContainer = <div className={carbonTileclassNames}>{cardContent}</div>;
+  } else if (isLink === true) {
+    cardContainer = (
+      <Link to={href} className={carbonTileclassNames} {...rest}>
+        {cardContent}
+      </Link>
+    );
+  } else {
+    cardContainer = (
+      <a href={href} className={carbonTileclassNames} {...rest}>
```

Comment:
if it's leaving the site, should we add a `target="_blank"` here? Otherwise, I'm not sure what the difference is between this and line 137.

Reply:
As I understand it, the else-if sets up the <Link> component so that when you navigate to a new page within the site, it does [preloading](https://www.gatsbyjs.org/docs/gatsby-link/) for a performance boost.. As opposed to the else which sets up the href, which will load the entire site.